src/components/ReportActionItem/ReportPreview.tsx

```diff
@@ -28,163 +23,134 @@ import * as ReceiptUtils from '@libs/ReceiptUtils';
-    iouReportID: PropTypes.string.isRequired,
-
-    /** The report's policyID, used for Onyx subscription */
-    policyID: PropTypes.string.isRequired,
+type PaymentVerbTranslationPath = 'iou.payerSpent' | 'iou.payerOwes' | 'iou.payerPaid';
 
+type ReportPreviewOnyxProps = {
     /** The policy tied to the money request report */
-    policy: PropTypes.shape({
-        /** Name of the policy */
-        name: PropTypes.string,
-
-        /** Type of the policy */
-        type: PropTypes.string,
+    policy: OnyxEntry<Policy>;
 
-        /** The role of the current user in the policy */
-        role: PropTypes.string,
+    /** ChatReport associated with iouReport */
+    chatReport: OnyxEntry<Report>;
 
-        /** Whether Scheduled Submit is turned on for this policy */
-        isHarvestingEnabled: PropTypes.bool,
-    }),
+    /** Active IOU Report for current report */
+    iouReport: OnyxEntry<Report>;
 
-    /* Onyx Props */
-    /** chatReport associated with iouReport */
-    chatReport: reportPropTypes,
+    /** Session info for the currently logged in user. */
+    session: OnyxEntry<Session>;
 
-    /** Extra styles to pass to View wrapper */
-    // eslint-disable-next-line react/forbid-prop-types
-    containerStyles: PropTypes.arrayOf(PropTypes.object),
+    /** All the transactions, used to update ReportPreview label and status */
+    transactions: OnyxCollection<Transaction>;
 
-    /** Active IOU Report for current report */
-    iouReport: PropTypes.shape({
-        /** AccountID of the manager in this iou report */
-        managerID: PropTypes.number,
+    /** All of the transaction violations */
+    transactionViolations: OnyxCollection<TransactionViolations>;
+};
 
-        /** AccountID of the creator of this iou report */
-        ownerAccountID: PropTypes.number,
+type ReportPreviewProps = ReportPreviewOnyxProps & {
+    /** All the data of the action */
+    action: ReportAction;
 
-        /** Outstanding amount in cents of this transaction */
-        total: PropTypes.number,
+    /** The associated chatReport */
+    chatReportID: string;
 
-        /** Currency of outstanding amount of this transaction */
-        currency: PropTypes.string,
+    /** The active IOUReport, used for Onyx subscription */
+    iouReportID: string;
 
-        /** Is the iouReport waiting for the submitter to add a credit bank account? */
-        isWaitingOnBankAccount: PropTypes.bool,
-    }),
+    /** The report's policyID, used for Onyx subscription */
+    policyID: string;
 
-    /** Session info for the currently logged in user. */
-    session: PropTypes.shape({
-        /** Currently logged in user accountID */
-        accountID: PropTypes.number,
-    }),
+    /** Extra styles to pass to View wrapper */
+    containerStyles?: StyleProp<ViewStyle>;
 
     /** Popover context menu anchor, used for showing context menu */
-    contextMenuAnchor: refPropTypes,
+    contextMenuAnchor?: ContextMenuAnchor;
```

Comment:
I think ContextMenuAnchor should be used in other places in the code base as well. It is usually named `anchor`